### PR TITLE
10117 Story: My Cases - Filing Fee Status Display

### PR DESCRIPTION
**Pull request by rachelschneiderman**
Petitioner (also applies for private practitioners) dashboard desktop:
Screenshot 2023-11-17 at 12 46 55 PM

Petitioner (also applies for private practitioners) dashboard mobile:
Screenshot 2023-11-17 at 12 46 43 PM

Petitioner (also applies for private practitioners) dashboard mobile (consolidated cases):
Screenshot 2023-11-17 at 12 46 33 PM

Petitioner (also applies for private practitioners) case details:
Screenshot 2023-11-17 at 12 47 04 PM

IRS practitioner dashboard desktop (unchanged, does not display filing fee):
image

IRS practitioner dashboard mobile (new mobile table display, does not display filing fee): 
image

### CODE REVIEW

File: cypress/cypress-integration/integration/petitioner-dashboard.cy.ts
Comment (mmarcotte) [inline cypress/cypress-integration/integration/petitioner-dashboard.cy.ts:1]: i'm just loving these. thank you!!

File: shared/src/business/useCases/getCasesForUserInteractor.ts [hunk 1]
```diff
@@ -21,26 +33,43 @@ export const getCasesForUserInteractor = async (
...
-  const allUserCases = await applicationContext
-    .getPersistenceGateway()
-    .getCasesForUser({
+  const docketNumbers = (
+    await applicationContext.getPersistenceGateway().getCasesForUser({
       applicationContext,
       userId,
-    });
+    })
+  ).map(c => c.docketNumber);
```
Comment (mmarcotte) [inline shared/src/business/useCases/getCasesForUserInteractor.ts:41]: interesting, so when working on this story did you just find the mapping records too difficult to maintain their information?

File: web-api/workflow-terraform/migration/main/lambdas/migrations/cleanup-usercase-mappings.ts
Comment (mmarcotte) [inline web-api/workflow-terraform/migration/main/lambdas/migrations/cleanup-usercase-mappings.ts:1]: Is there harm in preserving the stale information? Does this migration need to be run? Could we find these cases by querying Elasticsearch for the users, and then Dynamo for the cases? It's worth considering a solution that does not necessitate a migration.

Comment (rachelschneiderman) [reply web-api/workflow-terraform/migration/main/lambdas/migrations/cleanup-usercase-mappings.ts:1]: Chatted about this in Zoom and resolved with a one-time migration script!

**Review by mmarcotte — APPROVED** [review]

**Pull request merged**